Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -203,146 +203,180 @@ function xASL_wrp_ResampleASL(x)
 
 
 %% ------------------------------------------------------------------------------------------
-% 8. Pair-wise subtraction & saving PWI & PWI4D in both spaces
+% 8. Pair-wise control-label subtraction
+% for both native & standard space
+
+% Nomenclature:
+% ASL4D = original timeseries
+% PWI4D = subtracted timeseries -> for quantification purposes
+% PWI3D = averaged per TE—PLD—LabDur combination -> for QC purposes
+% PWI   = single perfusion-weighted volume -> for registration purposes
+
+% 
+
 % Load ASL time series (after being pre-processed)
 StringSpaceIs = {'native' 'standard'};
-PathASL = {x.P.Path_rdespiked_ASL4D x.P.Path_rtemp_despiked_ASL4D};
+PathASL4D = {x.P.Path_rdespiked_ASL4D x.P.Path_rtemp_despiked_ASL4D};
 PathPWI = {x.P.Path_PWI x.P.Pop_Path_PWI};
+PathPWI3D = {x.P.Path_PWI3D x.P.Pop_Path_PWI3D};
 PathPWI4D = {x.P.Path_PWI4D x.P.Pop_Path_PWI4D};

```

Comment:
I understand that of course, but we might need it (temporarily) for QC purposes. We can later improve this, or we should make QC images like SD in native space in general, and then move them to standard space (which may create a bit more artifacts but indeed be much faster).

Note that BASIL works in any space.

Reply:
Temporarily if really needed. But you still need to remove this or later delete and avoid saving that huge file.

Theoretically it works in any space of course :) But practically, it will never finish calculation :)

Reply:
I understand, and we don't have to run BASIL on it. But at least temporarily, we need to save it for the QC. All the JPG QC in the *check folders is in standard space, so it is visually easy to compare between patients etc, and we want this as well for some QC things that we compute from PWI4D and PWI3D. We can do this more efficiently, but then I would do this in a separate issue.

So I propose:

1. We keep this like it is
2. x.DELETETEMP — which defaults to true— will delete PWI4D (and PWI3D) in standard space, after the xASL_module_ASL
3. There will be a QC issue to compute all QC images from PWI3D and PWI4D in standard space
4. We will improve these QC computations by running them in native space and then transforming them to standard space, in a transparent way

Then it would be nice for generalization, that we also do quantification in native space. But the problem with that (similar to BASIL/FABBER now) is that motion correction and other interpolations are applied in the same native space, smoothing and loosing detail. That was the motivation for doing quantification in standard space.

BTW: in principle, fine to set (1) above already to false. But this would go against the standardization (all other QC is computed in standard space) so making the code more complex. And note that the BASIL/FABBER computations themselves already in native space take much more time than the transformation to standard space. This would be something to improve upon, by knowing the "resolution" at which we should do the fitting (is actually not needed voxel-wise) but this is tricky because this is spatially variant (non-local...).

Reply:
Do we really need PWI4D for QC? 
Because previously, we have managed doing all QC in single-PLD on the averaged image. And now, we will create huge PWI4D in standard space also for all single-PLD pCASLS. So it's not only the multi-PLD sequences that are affected, but also all single-PLDs will have a huge increase in processing and space requirements.
I know the standard space motivation for joint transformations, but that anyway doesn't work with BASIL.

Reply:
Yes. We want in standard space, NIfTIs and then JPGs, for:

* TE & PLD fit, so you can check if a fit goes wrong because of the data itself or an interaction with the DEBBIE model
* SD or difference between two DEBBIE iterations
* etc
* etc

We could do this all at the end of xASL_wrp_ResampleASL, but the point is that we want to give the user the ability to manually merge PWI4D or remove some pairs, etc, so then it is more transparent to keep PWI4D and delete it at the end of the xASL_module_ASL. We don't need to keep PWI4D for the population module. We only need PWI4D for e.g., merging, but that is native space. 

So easiest for now is to remove the standard space PWI4D at the end of xASL_module_ASL. Agreed?

Reply:
Yes. You've convinced me. You already implemented that, right? If yes, then please resolve.